app/Http/Controllers/Document/BulkDownloadController.php

```diff
@@ -0,0 +1,109 @@
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use KBox\Http\Controllers\Controller;
+use Zip;
+use KBox\Http\Requests\BulkDownloadRequest;
+use KBox\DocumentDescriptor;
+use Illuminate\Contracts\Auth\Guard as AuthGuard;
+use Illuminate\Support\Facades\Storage;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use ZipStream\ZipStream;
+use Illuminate\Http\JsonResponse;
+
+use function Matrix\trace;
+
+class BulkDownloadController extends Controller
+{
+        /*
+    |--------------------------------------------------------------------------
+    | Bulk Operation on Documents and Groups Controller
+    |--------------------------------------------------------------------------
+    |
+    | handle the operation when something is performed on a multiple selection.
+    | To simply JS stuff
+    |
+    */
+
+    /**
+     * [$service description]
+     * @var \KBox\Documents\Services\DocumentsService
+     */
+    private $service = null;
+
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct(\KBox\Documents\Services\DocumentsService $adapterService)
+    { 
+        $this->middleware('auth');
+
+        $this->service = $adapterService;
+    }
+
+    
+    //
+    public function buildzip(AuthGuard $auth, BulkDownloadRequest $request){
+
+        try {
+            \Log::info('Bulk  Download', ['params' => $request->all()]);
+            
+            //$user = $auth->user();
+            $docs = $request->input('documents', []);
+            $status = [];
+
+            $files_to_zip = [];
+
+                $docs = $request->input('documents', []);
+
+                $documents = DocumentDescriptor::whereIn('id', $docs)->with('file')->get();
+                foreach ($documents as $key => $value) {
+                    $files_to_zip=array_merge($files_to_zip,[$value->file->absolute_path => $value->file->name]);
+                }   
+
+                
+                $zipSize = Zip::create('download.zip',$files_to_zip)->predictZipSize()/1000000;
+               
+                if($zipSize <50 ){
+
+                    return Zip::create('download.zip',$files_to_zip)->response();  
+
+                }elseif ($zipSize > 50 &&  $zipSize < 4000){
+
+                    $status = [
+                        'status' => 'ok',
+                        'title' => trans('documents.bulk.download_file_size_over.title'),
+                        'message' => trans('documents.bulk.download_file_size_over.message')
+                    ];
+
+                }else {
+                    
+                    $status = [
+                        'status' => 'ok',
+                        'title' => trans('documents.bulk.download_file_size_big.title'),
+                        'message' => trans('documents.bulk.download_file_size_big.message')
+                    ];
+                } 
+
+                return response($status, 200)
```

Comment:
Is more appropriate to use a `JsonResponse` so you don't have to specify the content type